### PR TITLE
Angular app with login , new user Registration, dashboard and product…

### PR DESCRIPTION
**Pull request by KajalMathur**
Hello
Kindly review my PR.

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<mat-toolbar class="container" color="primary" style="flex-grow: 5">Shop Easy
+  <div class="logout_container">
+    <button (click)="login()" color="accent" mat-raised-button>Logout</button>
```
Comment (AakashIsTheLimit) [inline src/app/app.component.html]: logout button should not be calling `login` method. Maybe call it `navigateToLogin`

File: src/app/login-page/login-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  submit() {
+    if (!this.loginForm.valid)
+      return;
+    this.authService.authenticateUser(this.loginForm.value.userName, this.loginForm.value.password)
+      .subscribe((data) => {
+        this.router.navigate(['/dashboard'])
+      }, error => {
+        console.log("error = " + error.status);
```
Comment (AakashIsTheLimit) [inline src/app/login-page/login-page.component.ts]: error should be reported as `console.error`

File: src/app/user-registration/user-registration.component.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  saveUser() {
+    if (!this.signUpForm.valid)
+      return;
+    this.userService.createUser(this.signUpForm.value.firstName, this.signUpForm.value.lastName
+      , this.signUpForm.value.userName, this.signUpForm.value.password, this.signUpForm.value.address)
+      .subscribe((data) => {
+        this.router.navigate(['/dashboard']);
+      }, (error) => {
+        console.log("error = " + error.status);
```
Comment (AakashIsTheLimit) [inline src/app/user-registration/user-registration.component.ts]: console.error

File: src/app/user-registration/user-registration.component.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+@Component({
+  selector: 'app-user-registration',
+  templateUrl: './user-registration.component.html',
+  styleUrls: ['./user-registration.component.css']
+})
+export class UserRegistrationComponent implements OnInit {
+
+  signUpForm: FormGroup;
+
+
```
Comment (AakashIsTheLimit) [inline src/app/user-registration/user-registration.component.ts]: extra spaces

File: src/app/service/user/user.service.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+  
+  url = "http://localhost:9098/v1/user";
+
+  constructor(private httpClient: HttpClient) { }
+
+  public createUser(firstName: String, lastName: String, userName: String, password: string, address: Address) {
```
Comment (AakashIsTheLimit) [inline src/app/service/user/user.service.ts]: it should be public `async` . Read about now

File: src/app/service/product/product.service.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+  constructor(private httpClient: HttpClient) { }
+
+  public getAllProducts() {
+    console.log(this.url)
+    return this.httpClient.get(this.url);
+  }
+
+  public getProductById(userId: string, sellerDetail: boolean = false) {
+    return this.httpClient.get(this.url + "/" + userId + "?" + "sellerDetail=" + sellerDetail);
```
Comment (AakashIsTheLimit) [inline src/app/service/product/product.service.ts]: you should use `${this.url}/${userId}` like this in typescript

File: src/app/user-registration/user-registration.component.html [hunk 7]
```diff
@@ -0,0 +1,90 @@
+<div class="mat_container">
+  <mat-card class="matcard_size">
+
+    <form (submit)="saveUser()" [formGroup]="signUpForm" class="height">
```
Comment (AakashIsTheLimit) [inline src/app/user-registration/user-registration.component.html:4]: Read more about reactive forms in angular. 
Add errors to each of the fields : https://medium.com/@vivek0208/create-form-using-angular-material-f2472dab1687

look for "<mat-error>" fields added to each of the fields in the form.

File: src/app/seller-info/seller-info.component.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+@Component({
+  selector: 'app-seller-info',
+  templateUrl: './seller-info.component.html',
+  styleUrls: ['./seller-info.component.css']
+})
+export class SellerInfoComponent implements OnInit {
+
+  @Input()
+  product:Product;
```
Comment (AakashIsTheLimit) [inline src/app/seller-info/seller-info.component.ts]: after colon there should be space

File: src/app/app.component.html (hunk 1 above)
Comment (Kajal-techy) [reply src/app/app.component.html]: Done.

File: src/app/login-page/login-page.component.ts (hunk 2 above)
Comment (Kajal-techy) [reply src/app/login-page/login-page.component.ts]: Done.

File: src/app/user-registration/user-registration.component.ts (hunk 3 above)
Comment (Kajal-techy) [reply src/app/user-registration/user-registration.component.ts]: Done.

File: src/app/user-registration/user-registration.component.ts (hunk 4 above)
Comment (Kajal-techy) [reply src/app/user-registration/user-registration.component.ts]: Removed.

File: src/app/service/user/user.service.ts (hunk 5 above)
Comment (Kajal-techy) [reply src/app/service/user/user.service.ts]: done.

File: src/app/service/product/product.service.ts (hunk 6 above)
Comment (Kajal-techy) [reply src/app/service/product/product.service.ts]: Done.

File: src/app/user-registration/user-registration.component.html (hunk 7 above)
Comment (Kajal-techy) [reply src/app/user-registration/user-registration.component.html:4]: Done.

File: src/app/seller-info/seller-info.component.ts (hunk 8 above)
Comment (Kajal-techy) [reply src/app/seller-info/seller-info.component.ts]: Done.

**Review by AakashIsTheLimit — APPROVED** [review]

**Pull request merged**